### PR TITLE
feat: node 20.x update

### PR DESCRIPTION
**Pull request by capJavert**
Update to Node 20.x as latest LTS. Will merge after deps update. Checked locally and containers run fine.

### CODE REVIEW

**Review by omBratteng — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -103,7 +103,7 @@ jobs:
   pulumi_up:
     circleci_ip_ranges: true
     docker:
-      - image: cimg/node:18.17
+      - image: cimg/node:20.12
```
Comment (rebelchris) [inline .circleci/config.yml:106]: These match node alpine?
Maybe we should set the nvmrc one to match explicitly?

**Review by rebelchris — APPROVED** [review]
Non blocking comments

**Review by idoshamun — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (capJavert) [reply .circleci/config.yml:106]: Yeah, alpine always runs its own version which is updated as the upstream image is updated. Minor version of node won't be breaking but yeah we can probably update .nvmrc to match.

Comment (rebelchris) [reply .circleci/config.yml:106]: Yeah in apps I set everything to 20.12 just to be more explicit.
(non blocking though)

**Pull request merged**